instance/instance.go

```diff
@@ -53,32 +54,57 @@ const (
 //GetList a list of all instances
 func (s *Store) GetList(w http.ResponseWriter, r *http.Request) {
 	ctx := r.Context()
-	stateFilterQuery := r.URL.Query().Get("state")
-	var stateFilterList []string
-	if stateFilterQuery != "" {
-		stateFilterList = strings.Split(stateFilterQuery, ",")
-		if err := models.ValidateStateFilter(stateFilterList); err != nil {
-			log.ErrorCtx(ctx, err, nil)
-			http.Error(w, err.Error(), http.StatusBadRequest)
-			return
-		}
-	}
+	data := log.Data{}
+	var stateFilterQuery string
+	var ap common.Params = nil
 
-	results, err := s.GetInstances(stateFilterList)
-	if err != nil {
-		log.ErrorCtx(ctx, err, nil)
+	if err := s.Auditor.Record(ctx, GetInstancesAction, audit.Attempted, nil); err != nil {
```

Comment:
Think about using `ap` instead of `nil` here.

The case for using `ap` is "in the future we may add an audit parameter prior to this call to Auditor".

Reply:
I didn't see the point in using `ap` here as it's always empty at this point. In this scenario auditing the attempted action has failed before any business logic is carried out. I thought about pulling out the query parameter before the audit but it felt wrong to have some business logic separated from the rest.

Reply:
Im not convinced, I would take the stance on that information was part of the request and should be part of the audit and if the audit failed should be in the audit log.

I feel like a third opinion might help.